docs/debugging-testing/testing/unit_testing.md

```diff
@@ -61,19 +61,21 @@ In a terminal window:
 
 ## Test code structure
 
-Find unit tests in the Mbed OS repository under the `UNITTESTS` folder of each library. We recommend unit test files use an identical directory path as the file under test. This makes it easier to find unit tests for a particular class or a module. For example, if the file you're testing is `some/example/path/ClassName.cpp`, then all the test files are in the `UNITTESTS/some/example/path/ClassName` directory. Each test suite needs to have its own `unittest.cmake` file for test configuration.
+Find unit tests in the Mbed OS repository under the `UNITTESTS` folder of each library. We recommend unit test files use an identical directory path as the file under test. This makes it easier to find unit tests for a particular class or a module. For example, if the file you're testing is `some/example/path/ClassName.cpp`, then all the test files are in the `UNITTESTS/some/example/path/ClassName` directory. Each test suite needs to have its own `CMakeLists.txt` file for test CMake configuration.
 
-All the class stubs should be located in the Mbed OS root directory `UNITTESTS/stubs`. Multiple test suites can use a single stub class, which should follow the naming convention `ClassName_stub.cpp` for the source file and `ClassName_stub.h` for the header file. Use the actual header files for the unit tests, and don't stub headers if possible. The stubbed headers reside in the `UNITTESTS/target_h` directory.
+All the stub sources are grouped into stub libraries `mbed-stubs-rtos`, `mbed-stubs-drivers`, `mbed-stubs-events`, `mbed-stubs-hal`, `mbed-stubs-platform`, `mbed-stubs-storage`, `mbed-stubs-connectivity` and all stub libraries are target linked under `mbed-stubs`. The library unit test suite CMake is expected to link with the required stub library or directly link with mbed-stubs depends on their test cases.
 
-### Test discovery
+The new stub file should follow the naming convention `ClassName_stub.cpp` for the source file and `ClassName_stub.h` for the header file and add it under respective existing stub library or create their stubs Libary depends on their test cases.
+
+All the Mbed OS headers are grouped into header interface libraries like, `mbed-headers-platform`, `mbed-headers-connectivity`, `mbed-headers-storage`, `mbed-headers-drivers`, `mbed-headers-hal`, `mbed-headers-events`, `mbed-headers-rtos` and stubbed headers reside in the `UNITTESTS/target_h` into  `mbed-headers-base` library and all headers libraries are target linked under `mbed-headers`. The library unit test suite is a link with the required header library or directly link with mbed-headers depends on their test cases.
```

Comment:
```suggestion
All the Mbed OS header files are built with CMake `INTERFACE` libraries (e.g`mbed-headers-platform`). Stubbed header files reside in the `UNITTESTS/target_h` and are built with the `mbed-headers-base` CMake library. All CMake libraries containing header files are linked with `mbed-headers`. The CMake target of the library unit under test is expected to link with the required header file libraries or `mbed-headers` depending on the test case. 
```

`...depending on the test case. `  , can you give examples when to do one or the other?

Reply:
In case of the file under test uses headers that are not part of mbed os then  It requires creating a new header library  to provide that header implementation

Reply:
Please add your explanation in the readme.